actions/avatar/crop.php

```diff
@@ -12,58 +12,24 @@
 	forward(REFERER);
 }
 
-$x1 = (int) get_input('x1', 0);
-$y1 = (int) get_input('y1', 0);
-$x2 = (int) get_input('x2', 0);
-$y2 = (int) get_input('y2', 0);
-
-$filehandler = new ElggFile();
-$filehandler->owner_guid = $owner->getGUID();
-$filehandler->setFilename("profile/" . $owner->guid . "master" . ".jpg");
-$filename = $filehandler->getFilenameOnFilestore();
+$coords = [
+	'x1' => (int) get_input('x1', 0),
+	'y1' => (int) get_input('y1', 0),
+	'x2' => (int) get_input('x2', 0),
+	'y2' => (int) get_input('y2', 0),
+];
 
 // ensuring the avatar image exists in the first place
-if (!file_exists($filename)) {
+if (!$owner->hasIcon('master')) {
 	register_error(elgg_echo('avatar:crop:fail'));
 	forward(REFERER);
 }
 
-$icon_sizes = elgg_get_config('icon_sizes');
-unset($icon_sizes['master']);
-
-// get the images and save their file handlers into an array
-// so we can do clean up if one fails.
-$files = array();
-foreach ($icon_sizes as $name => $size_info) {
-	$resized = get_resized_image_from_existing_file($filename, $size_info['w'], $size_info['h'], $size_info['square'], $x1, $y1, $x2, $y2, $size_info['upscale']);
-
-	if ($resized) {
-		//@todo Make these actual entities.  See exts #348.
-		$file = new ElggFile();
-		$file->owner_guid = $guid;
-		$file->setFilename("profile/{$guid}{$name}.jpg");
-		$file->open('write');
-		$file->write($resized);
-		$file->close();
-		$files[] = $file;
-	} else {
-		// cleanup on fail
-		foreach ($files as $file) {
-			$file->delete();
-		}
-
-		register_error(elgg_echo('avatar:resize:fail'));
-		forward(REFERER);
-	}
+if (!$owner->saveIconFromElggFile($owner->getIcon('master'), 'icon', $coords)) {
```

Comment:
Something wiped `_uploaded` somewhere along the lines. Will need to update the tests to check what's happening.